contracts/utils/Arrays.sol

```diff
@@ -23,55 +23,151 @@ library Arrays {
      * when executing this as part of a transaction. If the array being sorted is too large, the sort operation may
      * consume more gas than is available in a block, leading to potential DoS.
      */
+    function sort(
+        bytes32[] memory array,
+        function(bytes32, bytes32) pure returns (bool) comp
+    ) internal pure returns (bytes32[] memory) {
+        uint256 begin;
+        uint256 end;
+        /// @solidity memory-safe-assembly
+        assembly {
+            begin := add(array, 0x20)
+            end := add(begin, mul(mload(array), 0x20))
+        }
+        _quickSort(begin, end, comp);
+        return array;
```

Comment:
An alternative would be
```suggestion
        uint256 ptr;
        /// @solidity memory-safe-assembly
        assembly {
            ptr := add(array, 0x20)
        }
        unchecked {
            _quickSort(ptr, ptr + array.length * 0x20, comp);
        }
        return array;
```

Reply:
I prefer the current state. If we're trying to preserver readability I think the original is best. Although I'd prefer a non-assembly alternative